### PR TITLE
feat(db): allow primary key passed as full entity to DAOs

### PR DESCRIPTION
**Pull request by bungle**
### Summary

Previously you needed to write code like this:
```lua
local route = kong.db.routes:select_by_name("my-route")
kong.db.routes:update({ id = route.id }, {
  paths = { "/test" },
})
kong.db.routes:delete({ id = route.id })
```

with this change you can write it like this:

```lua
local route = kong.db.routes:select_by_name("my-route")
kong.db.routes:update(route, {
  paths = { "/test" },
})
kong.db.routes:delete(route)
```
### Checklist

- [x] The Pull Request has tests
- [x] A changelog file has been added to `CHANGELOG/unreleased/kong` or adding `skip-changelog` label on PR if unnecessary. [README.md](https://github.com/Kong/kong/blob/master/CHANGELOG/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### CODE REVIEW

**Comment by bungle** [comment]
A follow up for this PR would be to allow foreign keys to be given as full entities, like this:

Before:
```lua
local service = kong.db.services:insert({ url = "http://test.test/" })
local route = kong.db.routes:insert({
  paths = { "/test" },
  service = { id = service.id },
})
```

After:
```lua
local service = kong.db.services:insert({ url = "http://test.test/" })
local route = kong.db.routes:insert({
  paths = { "/test" },
  service = service,
})
```

**Review by nowNick — APPROVED** [review]
Awesome 🚀

File: spec/02-integration/03-db/08-declarative_spec.lua [hunk 1]
```diff
@@ -239,7 +239,7 @@ for _, strategy in helpers.each_strategy() do
         assert.equals(sni_def.id, sni.id)
         assert.equals(certificate_def.id, sni.certificate.id)
 
-        local cert = assert(db.certificates:select({ id = certificate_def.id }))
+        local cert = assert(db.certificates:select(certificate_def))
```
[the remark this answers is not in the document]
Comment (bungle) [reply spec/02-integration/03-db/08-declarative_spec.lua:242]: Both are supported. Our tests seem to already use both.

**Review by samugi — APPROVED** [review]
looks good to me

**Pull request merged**